File: Justin_folder/sqs_queue.py
import boto3
import time

QUEUE_URL = "https://sqs.us-east-1.amazonaws.com/576607007622/gpsInfoQueue"
REGION = "us-east-1"
POLL = 5 #polls every 5 seconds

def main():
    #Create SQS client
    sqs = boto3.client("sqs", region_name=REGION)
    print("Listening for messages from SQS...")
    while True:
        try:
            #Receive messages (10 at once)
            response = sqs.receive_message(QueueUrl=QUEUE_URL, MaxNumberOfMessages=10, WaitTimeSeconds=10)
            messages = response.get("Messages", [])
            if not messages:
                print("No messages. Waiting...")
            else:
                for msg in messages:
                    body = msg["Body"]
                    receipt_handle = msg["ReceiptHandle"]

                    #Process the message
                    print(f"Received: {body}")
                    #Delete message from queue after processing
                    sqs.delete_message(QueueUrl=QUEUE_URL, ReceiptHandle=receipt_handle)
                    print("Message deleted.")
        except Exception as e:
            print(f"Error: {e}")

        #Pause before polling again
        time.sleep(POLL)

if __name__ == "__main__":
    main()

File: Justin_folder/sqs_queue.js
import{SQSClient, ReceiveMessageCommand, DeleteMessageCommand} from "@aws-sdk/client-sqs";

const QUEUE_URL = "https://sqs.us-east-1.amazonaws.com/576607007622/gpsInfoQueue"
const REGION = "us-east-1"
const POLL = 5 //polls every 5 seconds

//Create sqs client
const sqs = new SQSClient({region:REGION})
async function pollMessages() {
    console.log("Listening for messages from SQS...");
    while(true){
        try{
            //Receive 10 messages (at once)
            const response = await sqs.send(new ReceiveMessageCommand({QueueUrl:QUEUE_URL, MaxNumberOfMessages:10, WaitTimeSeconds:10,}));
            const messages = response.Messages || [];

            if(messages.length == 0){
                console.log("No messages. Waiting...");
            } else{
                for(const msg of messages){
                    const body = msg.Body;
                    const receiptHandle = msg.ReceiptHandle;
                    console.log('Received: ${body}'); //processing

                    //Delete messages from queue after processing
                    await sqs.send(new DeleteMessageCommand({QueueUrl:QUEUE_URL, ReceiptHandle:receiptHandle,}));
                    console.log("Messages deleted.");
                }
            }
        } catch(err){
            console.log("Error: ", err)
        }

        //Add a pause before polling again
        await new Promise(resolve=>setTimeout(resolve, POLL));
    }
}

pollMessages();